python/ray/data/block.py

```diff
@@ -370,7 +370,17 @@ def batch_to_block(batch: DataBatch) -> Block:
 
                 # TODO(ekl) once we support Python objects within Arrow blocks, we
                 # don't need this fallback path.
-                return pd.DataFrame(dict(batch))
+
+                pd_batch = dict()
+                for key, value in batch.items():
+                    if isinstance(value, np.ndarray) and value.ndim > 1:
+                        pd_batch[key] = np.array(
+                            [",".join(value.flatten())], dtype=value.dtype
+                        )
```

Comment:
If we don't like this hack we can also just `raise ValueError("multi-dimension array not supported in batch {}", batch)`